app/controllers/application_controller.rb

```diff
@@ -5,6 +5,7 @@ class ApplicationController < ActionController::Base
 
   prepend_before_filter :restrict_iframes
   before_filter :set_cache_headers # Issue #1213, prevent cart emptying via cache when using back button
+  before_filter :set_show_cookie_banner
```

Comment:
Can we move this into a helper instead of adding code to the application controller? I don't think you need to persist the state and introduce a new variable.

Reply:
Sorry for not being clear. I meant to remove all your additions from the ApplicationController and add a helper like this:
```ruby
def show_cookie_banner?
  request.cookies['cookies_consent'].nil?
end
```

I find that a lot simpler than callbacks and an instance variable. But I'm also not the biggest rails pro. Are there other opinions, @Matt-Yorkley @sauloperez?

Reply:
Personally, I'm not a fan of callbacks either and I dislike having ivars in views. Both lead to a global state that is very hard to manage in the long run.

I would go with @mkllnk suggestion and call that `show_cookie_banner?` straight from the view.

Reply:
awesome, thanks for the advise guys. I have removed everything from applicationcontroller and also removed a few things from api/cookies_consent_controller and added the cookies logic to the new and beautiful cookies_consent_helper.

It feels this helper is ok like this BUT it's quite obvious this should not be a helper, it should be a Service. Where would our service layer live? In this specific case, for example, where would a service like CookiesConsentService be in terms of folders?
Thanks!

Reply:
There is currently the choice between `app/services` and `lib`. We haven't agreed yet which approach is better. And while you are right that application logic should go into a service object, I find it okay to leave a one-liner in the helpers. You need the helper anyway to call the service object and pass on the `request` variable.